packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,939 @@
+
+		const setpointTypes: HumidityControlSetpointType[] =
+			this.getValueDB().getValue(
+				getSupportedSetpointTypesValueID(this.endpointIndex),
+			) ?? [];
+
+		// Query each setpoint's current value
+		for (const type of setpointTypes) {
+			const setpointName = getEnumMemberName(
+				HumidityControlSetpointType,
+				type,
+			);
+			// Every time, query the current value
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message: `querying current value of setpoint ${setpointName}...`,
+				direction: "outbound",
+			});
+			const setpoint = await api.get(type);
+			if (setpoint) {
+				const logMessage = `received current value of setpoint ${setpointName}: ${
+					setpoint.value
+				} ${setpoint.scale.unit ?? ""}`;
+				this.driver.controllerLog.logNode(node.id, {
+					endpoint: this.endpointIndex,
+					message: logMessage,
+					direction: "inbound",
+				});
+			}
+		}
+	}
+}
+
+interface HumidityControlSetpointCCSetOptions extends CCCommandOptions {
+	setpointType: HumidityControlSetpointType;
+	value: number;
+	scale: number;
+}
+
+@CCCommand(HumidityControlSetpointCommand.Set)
+export class HumidityControlSetpointCCSet extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| HumidityControlSetpointCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.setpointType = options.setpointType;
+			this.value = options.value;
+			this.scale = options.scale;
+		}
+	}
+
+	public setpointType: HumidityControlSetpointType;
+	public value: number;
+	public scale: number;
+
+	public serialize(): Buffer {
+		const override =
+			this.getNodeUnsafe()?.deviceConfig?.compat?.overrideFloatEncoding;
+		this.payload = Buffer.concat([
+			Buffer.from([this.setpointType & 0b1111]),
+			encodeFloatWithScale(this.value, this.scale, override),
+		]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const scale = getScale(this.driver.configManager, this.scale);
+		return {
+			...super.toLogEntry(),
+			message: {
+				"setpoint type": getEnumMemberName(
+					HumidityControlSetpointType,
+					this.setpointType,
+				),
+				value: `${this.value} ${scale.unit}`,
+			},
+		};
+	}
+}
+
+@CCCommand(HumidityControlSetpointCommand.Report)
+export class HumidityControlSetpointCCReport extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		this._type = this.payload[0] & 0b1111;
+		if (this._type === 0) {
+			// Not supported
+			this._value = 0;
+			this._scale = getScale(this.driver.configManager, 0);
+			return;
+		}
```

Comment:
I'm not sure about this. Does this happen?

Reply:
There's no setpoint type 0 defined in the spec, I guess this is just a guard to ensure we have a sane state if a device returns type 0. Like with the other places this is handled, this comes from Thermostat Setpoint CC which works the same (no type 0). Happy to change if you want, but how should this be handled?

Reply:
Guess it doesn't hurt to keep this in for now. Maybe add a comment that this is a fallback for behavior that's not defined in the specs.

Reply:
```suggestion
		// Setpoint type 0 is not defined in the spec, prevent devices from using it.
		if (this._type === 0) {
			// Not supported
			this._value = 0;
			this._scale = getScale(this.driver.configManager, 0);
			return;
		}
```